tmt/steps/execute/__init__.py

```diff
@@ -185,6 +185,14 @@ def data_path(
         path = directory / filename
         return path if full else path.relative_to(self.phase.step.workdir)
 
+    @tmt.utils.cached_property
+    def check_files_path(self) -> Path:
+        """ Construct a directory path for check files needed by tmt """
+        path = self.data_path(create=True, full=True).joinpath("checks")
+        if not path.is_dir():
+            path.mkdir()
```

Comment:
* `path1 / path2` instead of `path1.joinpath(path2)`, please
* I wouldn't bother with `is_dir` & call `path.mkdir(exists=True)` right away. No other part of code will create the path for us, and this property won't get called again anyway, so the first time `path.is_dir()` would report "nope", and we would never get to `is_dir()` telling us "yep, lucky you, you're going to save one mkdir call!" because of the cached property.

Reply:
It can be a string, `Path` is smart and it would join nicely, producing new `Path` instance.